### ISSUE TITLE
Unable to use `resultEqualityCheck` with `weakMapMemoize`

### ISSUE THREAD

**Issue by psychedelicious** [reduxjs/reselect#679]
I'm unable to use `resultEqualityCheck` with `weakMapMemoize`. Runtime error:

```
TypeError
lastResult?.deref is not a function
memoized
https://gtqrl5.csb.app/node_modules/reselect/dist/cjs/reselect.cjs:620:43
dependenciesChecker
https://gtqrl5.csb.app/node_modules/reselect/dist/cjs/reselect.cjs:711:39
memoized
https://gtqrl5.csb.app/node_modules/reselect/dist/cjs/reselect.cjs:615:21
memoized
https://gtqrl5.csb.app/node_modules/react-redux/dist/cjs/react-redux.development.cjs:134:28
memoizedSelector
https://gtqrl5.csb.app/node_modules/use-sync-external-store/cjs/use-sync-external-store-with-selector.development.js:110:27
getSnapshotWithSelector
https://gtqrl5.csb.app/node_modules/use-sync-external-store/cjs/use-sync-external-store-with-selector.development.js:133:14
updateSyncExternalStore
https://gtqrl5.csb.app/node_modules/react-dom/cjs/react-dom.development.js:16860:22
Object.useSyncExternalStore
https://gtqrl5.csb.app/node_modules/react-dom/cjs/react-dom.development.js:17943:14
useSyncExternalStore
https://gtqrl5.csb.app/node_modules/react/cjs/react.development.js:1676:21
useSyncExternalStoreWithSelector
https://gtqrl5.csb.app/node_modules/use-sync-external-store/cjs/use-sync-external-store-with-selector.development.js:144:15
useSelector2
https://gtqrl5.csb.app/node_modules/react-redux/dist/cjs/react-redux.development.cjs:188:27
Counter
/src/features/counter/Counter.js:19:28
```

Minimal repro: https://codesandbox.io/p/sandbox/dazzling-kapitsa-gtqrl5?file=%2Fsrc%2Fapp%2Fstore.js%3A20%2C71

**Comment by aryaemami59** [comment]
This was fixed in #671. @EskiMojo14 I believe we have to release a 5.0.2 for that to take effect?

**Comment by EskiMojo14** [comment]
yep, I'll aim to get a release out today

**Comment by EskiMojo14** [comment]
Fix published in [v5.1.0](https://github.com/reduxjs/reselect/releases/tag/v5.1.0)!
